terraform/custom_domains/environment_domains/terraform.tf

```diff
@@ -0,0 +1,25 @@
+terraform {
+
+  required_version = "= 1.3.5"
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "3.45.0"
+    }
+  }
+  backend "azurerm" {
+  }
+}
+
+provider "azurerm" {
+  features {}
+
+  skip_provider_registration = true
+}
+
+provider "azurerm" {
+  features {}
+  version         = "3.45.0"
```

Comment:
This is redundant with the required_providers configuration and [deprecated](https://developer.hashicorp.com/terraform/language/providers/configuration#version-deprecated)